libraries/AP_DroneCAN/AP_DroneCAN.cpp

```diff
@@ -1056,12 +1056,19 @@ void AP_DroneCAN::gnss_send_yaw()
 // SafetyState send
 void AP_DroneCAN::safety_state_send()
 {
-    uint32_t now = AP_HAL::millis();
-    if (now - _last_safety_state_ms < 500) {
-        // update at 2Hz
+    const uint32_t now_ms = AP_HAL::millis();
+    const uint32_t last_armed_change_ms = hal.util->get_last_armed_change();
+    const uint32_t interval_ms = (last_armed_change_ms <= 1000) ? 100 : 500;
```

Comment:
i think this should be:
  const uint32_t interval_ms = ((now_ms - last_armed_change_ms) <= 1000) ? 100 : 500;
?

Reply:
yup, that works too.